### PR TITLE
Delete post functionality added

### PR DESCRIPTION
**Pull request by Ajdj100**
A new endpoint for deleting posts has been added, and a button was added to your profile page which allows you to delete your own posts while browsing them. It may be too easy right now to delete a post and might be accidentally hit. Could be worth to add some sort of confirmation in the future.

### CODE REVIEW

**Comment by Ajdj100** [comment]
this should close #70

File: Public/Profile.html [hunk 1]
```diff
@@ -122,12 +136,25 @@
...
-            const body = `<div class="article" style="cursor: pointer;" onclick="navigateTo('./Article.html?id=${element.post_id}')"><h2>${element.title}</h2><h4>${element.username}</h4><p>${content}</p></div>`
+            const body = `<div class="article" style="cursor: pointer;"><div class="articleBody" onclick="navigateTo('./Article.html?id=${element.post_id}')"><h2>${element.title}</h2><h4>${element.username}</h4><p>${content}</p></div> <i class="bi bi-x-lg deleteButton" style="font-size: 2rem; cursor: pointer;" onclick="deletePost(${element.post_id})"></i></div>`
             let row = table.insertRow();
             row.innerHTML = body;
         });
     }
 
+    function deletePost(postID) {
+        //TODO: Implement delete fetch call
+        fetch("http://localhost:8000/post", {
```
Comment (sebastian-j-ibanez) [inline Public/Profile.html:147]: I do not believe this route exists yet. The only DELETE route I see is for "/", and is intended to delete comments.

File: Routes/post.js [hunk 2]
```diff
@@ -51,26 +51,58 @@ router.patch('/', function(req, res) {
...
+        if (error) {
+            res.status(500).json({ error: 'Error updating the post.' });
+        } else {
+            res.status(200).json({ message: 'Post updated successfully.' });
+        }
+    })
+});
+
+//Handling the edit post 
+router.delete('/', function (req, res) {
```
Comment (sebastian-j-ibanez) [inline Routes/post.js:76]: I think we should change this DELETE route from "/" to "/comment" to be more specific about what's being deleted.

**Review by sebastian-j-ibanez — CHANGES_REQUESTED** [review]
Added a few comments. Let me know what you think :mate:

File: Public/Profile.html (hunk 1 above)
Comment (sebastian-j-ibanez) [reply Public/Profile.html:147]: I misunderstood how the route seperation worked. This is properly implemented.

**Review by sebastian-j-ibanez — APPROVED** [review]
Looks good :mate:

**Pull request merged**